pkg/client/request.go

```diff
@@ -79,6 +79,14 @@ func (r *RequestConstructionError) Error() string {
 	return fmt.Sprintf("request construction error: '%v'", r.Err)
 }
 
+type PatchType string
```

Comment:
This enum is created and never used (except for once). I'd prefer if it was used in place of all the copy pasted string constants.